src/represents.rs

```diff
@@ -0,0 +1,152 @@
+        .parent_node()
+        .filter(|p| p.is_html_element(&local_name!("span")))?;
+    let p = span
+        .parent_node()
+        .filter(|p| p.is_html_element(&local_name!("p")))?;
+    let children = p.children.borrow();
+    match &children[..] {
+        [a, b, c, d, ..]
+            if a.node_text().as_deref().map(|x| x.trim()) == Some("The")
+                && b.is_html_element(&local_name!("code"))
+                && c.node_text().as_deref().map(|x| x.trim()) == Some("element")
+                && Rc::ptr_eq(d, &span) =>
+        {
+            Some((b.text_content(), span))
+        }
+        _ => None,
+    }
+}
+
+impl Processor {
+    pub fn new() -> Self {
+        Self {
+            represents: HashMap::new(),
+            placeholders: Vec::new(),
+        }
+    }
+
+    /// Should be called for each node the document. Records when it sees a
+    /// <span>represents</span> and which element it is defining
+    pub fn visit(&mut self, node: &Handle) {
+        match node.data {
+            NodeData::Text { ref contents } if contents.borrow().as_ref() == "represents" => {
+                if let Some((tag, span)) = find_tag_name(node) {
+                    self.represents.insert(tag, span);
+                }
+            }
+            NodeData::Comment { ref contents } if contents.starts_with("REPRESENTS ") => {
+                self.placeholders
+                    .push((node.clone(), contents.subtendril(11, contents.len32() - 11)));
+            }
+            _ => (),
+        }
+    }
+
+    pub fn apply(self) -> io::Result<()> {
+        for (placeholder, ref tag) in self.placeholders {
+            let span = match self.represents.get(tag) {
+                Some(span) => span,
+                None => {
+                    return Err(io::Error::new(
+                        io::ErrorKind::InvalidData,
+                        format!("<!--REPRESENTS {}--> refers to unknown tag", tag),
+                    ));
+                }
+            };
+            let parent = match span.parent_node() {
+                Some(p) => p,
+                None => continue,
+            };
+            let replacements = parent
+                .children
+                .borrow()
+                .iter()
+                .skip_while(|s| !Rc::ptr_eq(s, span))
+                .skip(1)
+                .enumerate()
+                .map(|(index, sibling)| {
+                    let clone = sibling.deep_clone();
+                    // Capitalize the first letter of the first node (which is expected to be text).
+                    if let (0, NodeData::Text { ref contents }) = (index, &clone.data) {
+                        contents.replace_with(|text| capitalize(text.trim_start()));
+                    }
+                    clone
+                })
+                .collect();
+            placeholder.replace_with(replacements);
+        }
+        Ok(())
+    }
+}
+
+fn capitalize(text: &str) -> StrTendril {
```

Comment:
`utils.rs`? (I guess the reason this isn't in the stdlib is because `StrTendril`s are a html5ever thing?)

Reply:
Actually afaik it doesn't exist in the Rust standard library either.